src/main/kotlin/pullpitok/App.kt

```diff
@@ -1,42 +1,53 @@
 package pullpitok
 
 import kotlin.system.exitProcess
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
 import pullpitok.github.Action
 import pullpitok.github.Event
 import pullpitok.github.EventClient
 import pullpitok.github.Type
 
-fun main(args: Array<String>) {
+suspend fun main(args: Array<String>) {
     if (invalidArguments(args)) {
         println(usage)
         exitProcess(0)
     }
     val repos = args[0].split(",")
     val token = args.getOrNull(1)
-    repos
-            .parallelStream()
-            .forEach { displayEvents(it, token) }
+    displayEvents(repos, token)
 }
 
 internal fun invalidArguments(args: Array<String>): Boolean =
         args.size !in (1..2) || args[0] in setOf("", "-h", "--help")
 
-private fun displayEvents(repo: String, token: String?) {
+private suspend fun displayEvents(repos: List<String>, token: String?) = coroutineScope {
+    val channel = Channel<Pair<String, List<Event>>>()
     val allEvents = mutableListOf<Event>()
-    for (pageNumber in 1..10) {
-        val events = EventClient().githubEvents(repo, token, page = pageNumber)
-        if (events.isNotEmpty()) allEvents.addAll(events)
-        else break
+    launch {
+        for (repo in repos) {
+            for (pageNumber in 1..10) {
+                val events = EventClient()
+                        .githubEvents(repo, token, page = pageNumber)
```

Comment:
TODO: check about failures ; it seems that default failure strategy is "print to console". 🤔 
See https://kotlinlang.org/docs/reference/coroutines/exception-handling.html#coroutineexceptionhandler.